### DISCUSSION TITLE
Kotlin SDK eth_signTypedData not working

### DISCUSSION THREAD

**Discussion by ruixhuang** [WalletConnect/walletconnect-monorepo#3044] [Developer Support] [answered]
Has anyone got eth_signTypedData working with wallet other than Trust Wallet?  I've tried Metamask, Zerion, Rainbow and none of them displays the signing dialog within the wallet app.  From the client side, the SignClient.request() call is successful, but the wallet doesn't show the request.   It's strange since the same payload works with Trust Wallet, but not the others.   Also, on the iOS client, eth_signTypedData works as expected with all wallets.

Payload: 
`["0x31Da8c137c4EB50a330885698A80c939Fe35Af8A", {"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"chainId","type":"uint256"}],"dYdX":[{"name":"action","type":"string"}]},"primaryType":"dYdX","domain":{"name":"dYdX","chainId":5},"message":{"action":"Sample Action"}}]`

SDK version: com.walletconnect:android-bom:1.13.1

**Comment by chris13524** [comment]
@TalhaAli00

**Comment by Talhaali00** [comment]
@ruixhuang

With the Android client, we have a test for `eth_signTypedData` [here](https://github.com/WalletConnect/WalletConnectKotlinV2/blob/23985daca3efa7d0ab2ffaf9f09da90a222f7a53/sign/sdk/src/test/kotlin/com/walletconnect/sign/core/adapters/SessionRequestVOJsonAdapterTest.kt#L237) and I was able to verify that the test still passes when using your payload. The test seems to corroborate what you're seeing on your end.

Have you had a chance to test your Android client with either and iOS or web wallet?

**Reply by ruixhuang** [reply]
Hi @TalhaAli00   Thanks for helping!

I've tested the Android client with iOS and Android wallets, and the behavior seems to be consistent regardless of the wallet platform.  `personal_sign` and `eth_sendTransaction` work fine, but not `eth_signTypedData`.  We are mainly trying to get Metamask to work.

The sample sign.app seems to get stuck after scanning the QR code to connect, and I haven't been able to test the transaction using the sample app.  Does it work on your end?

**Reply by chendaxia65** [reply]
@chris13524 @TalhaAli00 Thanks for helping!

I also encountered the same problem, using eth_signTypedData to sign, the result is {"code": -32603, "message": "Internal error"}, the same result is used in the example.

params=["0x0fee13F2DdD5ae14bF6347133cD37c5A059e15A5",{"domain":{"chainId":1,"name":"Login","version":"1"},"message":{"contents":"www.openvoice.com wants you to sign in with your Ethereum account:\n0x0fee13F2DdD5ae14bF6347133cD37c5A059e15A5\n\nSign in with Ethereum to the app.\n\nURI: https://www.openvoice.com\nVersion: 1\nChain ID: 1\nNonce: holhlmiDINJNJkyF\nIssued At: 2023-07-10T09:34:22.615Z"},"primaryType":"Login","types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"}],"Login":[{"name":"contents","type":"string"}]}}]

![20230710-153134](https://github.com/WalletConnect/walletconnect-monorepo/assets/19222274/75297d98-273b-4fef-b3e9-510ba6883042)

**Comment by Talhaali00** [comment, marked as answer]
@ruixhuang @chendaxia65 
We've released an update to better support eth_signedTypedData and eth_signedTypedDataV4 which are included in the latest releases. 

Can you please try out the SDKs and if you're still seeing an issue

**Reply by ruixhuang** [reply]
@TalhaAli00   It works now... Thanks for the fix!

@chendaxia65  Be sure to use escaped string as the payload params.  See [here](https://github.com/WalletConnect/WalletConnectKotlinV2/blob/develop/sample/common/src/main/kotlin/com/walletconnect/sample/common/Chains.kt#L21)
